### PR TITLE
Added Version Support to Dashboards API

### PR DESCRIPTION
**Pull request by cannakula-sfdc**
* Added Dashboard Version Support to /dashboards API 
* Added Dashboard  Version Support to /dashboards/meta API
* Added Dashboard  Version Support to /dashboards/shared API
* Added Dashboard  Version Support to /dashboards/shared/meta API

### CODE REVIEW

File: ArgusCore/src/main/java/com/salesforce/dva/argus/entity/Dashboard.java [hunk 1]
```diff
@@ -96,16 +94,16 @@
...
+        ), @NamedQuery(name = "Dashboard.getNewSharedDashboards", query = "SELECT d FROM Dashboard d WHERE d.shared = true AND d.version = :version"
+        ), @NamedQuery(name = "Dashboard.getOldDashboardsByOwner", query = "SELECT d FROM Dashboard d WHERE d.owner = :owner AND d.version IS NULL"
+        ), @NamedQuery(name = "Dashboard.getNewDashboardsByOwner", query = "SELECT d FROM Dashboard d WHERE d.owner = :owner AND d.version = :version"
+        ), @NamedQuery(name = "Dashboard.getOldDashboards", query = "SELECT d FROM Dashboard d WHERE d.version IS NULL ORDER BY d.owner.userName,d.name ASC"
+        ), @NamedQuery(name = "Dashboard.getNewDashboards", query = "SELECT d FROM Dashboard d WHERE d.version = :version ORDER BY d.owner.userName,d.name ASC"
+        ), @NamedQuery(name = "Dashboard.getOldSharedDashboardsByOwner",query = "SELECT d FROM Dashboard d WHERE d.owner = :owner AND d.shared = true AND d.version IS NULL"
+		), @NamedQuery(name = "Dashboard.getNewSharedDashboardsByOwner",query = "SELECT d FROM Dashboard d WHERE d.owner = :owner AND d.shared = true AND d.version = :version"
+        )
     }
```
Comment (bsura) [inline ArgusCore/src/main/java/com/salesforce/dva/argus/entity/Dashboard.java]: I would use meaningful names for the newly created named queries. For e.g.: Instead of "getNewSharedDashboards", maybe something like "getSharedDashboardsByVersion".

**Comment by bsura** [comment]
@cannakula-sfdc It would be great if you add test cases for the code path when version is not null.
